components/UI/Map/LeafletMap.tsx

```diff
@@ -65,21 +65,35 @@ const MapEvents = () => {
   const router = useRouter();
   const { setPopUpData, setEventType } = useMapActions();
   const { setCoordinates } = useURLActions();
+  const id = router.query["id"];
```

Comment:
you can use the next router for reaching router parameters.

Reply:
this is already the next router